be/src/vec/exec/volap_scan_node.cpp

```diff
@@ -2019,4 +1343,530 @@ Status VOlapScanNode::get_hints(TabletSharedPtr table, const TPaloScanRange& sca
     return Status::OK();
 }
 
+template <bool IsNotIn>
+bool VOlapScanNode::_should_push_down_in_predicate(VInPredicate* pred, VExprContext* expr_ctx) {
+    if (pred->is_not_in() != IsNotIn) {
+        return false;
+    }
+    InState* state = reinterpret_cast<InState*>(
+            expr_ctx->fn_context(pred->fn_context_index())
+                    ->get_function_state(FunctionContext::FRAGMENT_LOCAL));
+    HybridSetBase* set = state->hybrid_set.get();
+
+    // if there are too many elements in InPredicate, exceed the limit,
+    // we will not push any condition of this column to storage engine.
+    // because too many conditions pushed down to storage engine may even
+    // slow down the query process.
+    // ATTN: This is just an experience value. You may need to try
+    // different thresholds to improve performance.
+    if (set->size() > _max_pushdown_conditions_per_column) {
+        VLOG_NOTICE << "Predicate value num " << set->size() << " exceed limit "
+                    << _max_pushdown_conditions_per_column;
+        return false;
+    }
+    return true;
+}
+
+bool VOlapScanNode::_should_push_down_function(
+        VectorizedFnCall* fn_call, const std::function<bool(const std::string&)>& fn_checker) {
+    return fn_checker(fn_call->fn().name.function_name);
+}
+
+bool VOlapScanNode::_should_push_down_function_filter(VectorizedFnCall* fn_call,
+                                                      VExprContext* expr_ctx,
+                                                      std::string* constant_str,
+                                                      doris_udf::FunctionContext** fn_ctx) {
+    if (fn_call->fn().name.function_name != "like") {
```

Comment:
Add comment of now only support `like` function. wd do not support `not like` ?

Reply:
`not like` is supported now. This processing is included in `_normalize_function_filters`, e.g. a `COMPOUND_PRED` which is `not` will be processed first